business/business/domain/ports/realtime.py

```diff
@@ -0,0 +1,61 @@
+import abc
+from typing import Literal
+
+
+import rx.operators as op
+from rx.subject.subject import Subject
+
+from business.domain.models import events
+from business.domain.ports.domain_message_bus import AbstractDomainMessageBus
+
+RealtimeTopic = Literal["epci_action_statut_update"]
+
+epci_action_statut_update: RealtimeTopic = "epci_action_statut_update"
+
+
+class AbstractRealtime(abc.ABC):  # Possible name ? DataLayerObserver ?
+    """The Real time part of the data layer
+    Convert events from raw_source to typed events and push them into event_source"""
+
+    tables_to_observe = ["epci_action_statut_update"]
+
+    def __init__(self, domain_message_bus: AbstractDomainMessageBus):
+        self.external_observable = Subject()
+        # self._domain_observable = Subject()
+        self.domain_message_bus = domain_message_bus
+        self.external_observable.pipe(op.filter(self.table_filter)).subscribe(
```

Comment:
pourquoi ne pas remplacer publish to domain par un truc genre:
````python
        for converter in self.converters:
            self.external_observable.pipe(
                op.filter(converter.filter),
                op.map(converter.convert),
            ).subscribe(self.domain_message_bus.publish_event)
````
cf `data_layer/realtime.py`